app/src/main/java/com/example/pocketwatching/Activities/ProfileActivity.java

```diff
@@ -197,24 +210,81 @@ public void onFailure(Call<List<TxHistory>> call, Throwable t) {
+        for (int i = 0; i < times.size(); i++) {
+            getBlockHeight(times.get(i));
+        }
+
+        // for each wallet, get wallet balance at each block height in the past week in getBlockHeight
+    }
+
+    private void getEthPrices(String start, String end) {
+        Call<List<DateToBlock>> call = (Call<List<DateToBlock>>) PoloniexClient.getInstance().getPoloniexApi().getEthPrices(start, end);
+        call.enqueue(new Callback<List<DateToBlock>>() {
+            @Override
+            public void onResponse(Call<List<DateToBlock>> call, Response<List<DateToBlock>> response) {
+                Toast.makeText(ProfileActivity.this, "Successfully got historical eth prices", Toast.LENGTH_SHORT).show();
+            }
+
+            @Override
+            public void onFailure(Call<List<DateToBlock>> call, Throwable t) {
+                Toast.makeText(ProfileActivity.this, "Failed to get historical eth prices", Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
+
+    private void getBlockHeight(String timestamp) {
+        Call<DateToBlock> call = (Call<DateToBlock>) MoralisClient.getInstance().getMoralisApi().timeToBlock(timestamp);
         call.enqueue(new Callback<DateToBlock>() {
             @Override
             public void onResponse(Call<DateToBlock> call, Response<DateToBlock> response) {
-                getLastWeekBlock(timestamp, response.body().getBlock());
+                blockHeights.add(response.body().getBlock());
+                if (blockHeights.size() == 7) {
+                    Collections.sort(blockHeights);
+                    for (int i = 0; i < userWallets.size(); i++) {
+                        for (int j = 0; j < 7; j++) {
+                            getBlockBalance(userWallets.get(i).getWalletAddress(), blockHeights.get(j), j);
+                        }
+                    }
+                }
             }
 
             @Override
             public void onFailure(Call<DateToBlock> call, Throwable t) {
-                Toast.makeText(ProfileActivity.this, "Failed to access the API", Toast.LENGTH_SHORT).show();
+                Toast.makeText(ProfileActivity.this, "Could not get block from timestamp :(", Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
+
+    private void getBlockBalance(String address, int blockHeight, int index) {
+        Call<BlockBalance> call = (Call<BlockBalance>) MoralisClient.getInstance().getMoralisApi().getBlockBalance(address, blockHeight);
+        call.enqueue(new Callback<BlockBalance>() {
+            @Override
+            public void onResponse(Call<BlockBalance> call, Response<BlockBalance> response) {
+                blockBalances.set(index, Double.valueOf(response.body().getBalance()));
+                if (index == 6) {
+                    for (int i = 0; i < blockBalances.size(); i++) {
+                        // need to account for the value of eth at the time of the block
```

Comment:
ditto, add TODO if it's something you plan on implementing later that you want to keep track of.